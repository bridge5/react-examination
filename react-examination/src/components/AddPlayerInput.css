:local(.addPlayerInput) {
  border-radius: 0;
  border-color: #abaaaa;
  border-left: 0;
  border-right: 0;
}

:local(.addPlayerSelect) {
  border-radius: 0;
  border-color: #abaaaa;
  border-left: 0;
  border-right: 0;
}
