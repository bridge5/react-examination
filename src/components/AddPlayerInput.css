:local(.addPlayerInput) {
  border-radius: 0;
  border-color: #abaaaa;
  border-left: 0;
  border-right: 0;
}
.addPlayerInputBox {
  display: flex;
  width: 500px;
}
