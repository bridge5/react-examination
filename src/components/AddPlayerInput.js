import React, {Component} from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import styles from './AddPlayerInput.css';
import PlayerPositionSelect from "./PlayerPositionSelect";

class AddPlayerInput extends Component {
  render() {
    return (
      <>
        <input
          type="text"
          autoFocus={true}
          className={classnames('form-control', styles.addPlayerInput)}
          placeholder="Type the name of a player"
          value={this.state.name}
          onChange={this.handleChange.bind(this)}
          onKeyDown={this.handleSubmit.bind(this)}
        />
        <PlayerPositionSelect
          value={this.state.position}
          onChange={this.handleSelectChange}/>
      </>
    );
  }

  constructor(props, context) {
    super(props, context);
    this.state = {
      name: this.props.name || '',
      position: "SF"
    };
  }

  handleSelectChange = (value) => {
    this.setState({position: value});
  };

  handleChange(e) {
    this.setState({name: e.target.value});
  }

  handleSubmit(e) {
    const name = e.target.value.trim();
    if (e.which === 13) {
      const {position} = this.state;
      this.props.addPlayer(name,position);
      this.setState({name: ''});
    }
  }
}

AddPlayerInput.propTypes = {
  addPlayer: PropTypes.func.isRequired,
};

export default AddPlayerInput;
