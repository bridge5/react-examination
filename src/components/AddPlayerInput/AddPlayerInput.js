import React, { Component } from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import styles from './AddPlayerInput.css';

class AddPlayerInput extends Component {
  render() {
    return (
      <input
        type="text"
        autoFocus={true}
        className={classnames('form-control', styles.addPlayerInput)}
        placeholder="Type the name of a player"
        value={this.state.name}
        onChange={this.handleChange.bind(this)}
        onKeyDown={this.handleSubmit.bind(this)}
      />
    );
  }

  constructor(props, context) {
    super(props, context);
    this.state = {
      name: this.props.name || '',
    };
  }

  handleChange(e) {
    this.setState({ name: e.target.value });
  }

  handleSubmit(e) {
    const name = e.target.value.trim();
    if (e.which === 13) {
      this.props.addPlayer(name);
      this.setState({ name: '' });
    }
  }
}

AddPlayerInput.propTypes = {
  addPlayer: PropTypes.func.isRequired,
};

export default AddPlayerInput;
