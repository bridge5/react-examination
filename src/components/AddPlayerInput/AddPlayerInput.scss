:local(.addPlayerInput) {
  border-radius: 0;
  border-color: #abaaaa;
  border-left: 0;
  border-right: 0;
  margin: 5px 0;
}
:local(.playPositionSelector) {
  margin: 5px 0;
}

:local(.wrap) {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  .btn {
    margin: 5px;
  }
}
