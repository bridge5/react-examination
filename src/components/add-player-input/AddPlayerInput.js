import React, { Component } from "react";
import classnames from "classnames";
import PropTypes from "prop-types";
import { POS_TYPES, Options } from "../../constants/posTypes";

import styles from "./AddPlayerInput.module.scss";

class AddPlayerInput extends Component {
  render() {
    return (
      <div className={styles.addPlayerFeild}>
        <input
          type="text"
          autoFocus={true}
          className={classnames("form-control", styles.addPlayerInput)}
          placeholder="Type the name of a player"
          value={this.state.name}
          onChange={this.handleChange.bind(this)}
          onKeyDown={this.handleSubmit.bind(this)}
        />
        <select onChange={this.handlePositionSelect}>{Options}</select>
      </div>
    );
  }

  constructor(props) {
    super(props);
    this.state = {
      name: "",
      position: POS_TYPES.SF,
    };
  }

  handleChange(e) {
    this.setState({ name: e.target.value });
  }
  handlePositionSelect = (e) => {
    this.setState({ position: e.target.value });
  };
  handleSubmit(e) {
    const name = e.target.value.trim();
    if (e.which === 13) {
      console.log(this.state.position);
      this.props.addPlayer({ name, position: this.state.position });
      this.setState({ name: "" });
    }
  }
}

AddPlayerInput.propTypes = {
  addPlayer: PropTypes.func.isRequired,
};

export default AddPlayerInput;
