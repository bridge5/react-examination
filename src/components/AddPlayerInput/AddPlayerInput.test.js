import React from 'react';
import AddPlayerInput from './AddPlayerInput';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';

it('renders without crashing', () => {
  const spy = jest.fn();
  const wrapper = renderer.create(<AddPlayerInput addPlayer={spy} />).toJSON();
  expect(wrapper).toMatchSnapshot();
});

it('should trigger fn when button clicked', () => {
  const spy = jest.fn();
  const wrapper = mount(<AddPlayerInput addPlayer={spy} />);
  const button = wrapper.find('button').at(0);
  button.simulate('click');
  expect(spy.mock.calls.length).toBe(1);
});
