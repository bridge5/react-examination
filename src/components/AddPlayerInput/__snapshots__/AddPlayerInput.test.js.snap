// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <input
    autoFocus={true}
    className="form-control"
    onChange={[Function]}
    placeholder="Type the name of a player"
    type="text"
    value=""
  />
  <input
    autoFocus={true}
    className="form-control"
    onChange={[Function]}
    placeholder="Type the name of team"
    type="text"
    value=""
  />
  <select
    className="form-control"
    onChange={[Function]}
    value="SF"
  >
    <option
      value="SF"
    >
      SF
    </option>
    <option
      value="PF"
    >
      PF
    </option>
    <option
      value="C"
    >
      C
    </option>
    <option
      value="PG"
    >
      PG
    </option>
    <option
      value="SG"
    >
      SG
    </option>
  </select>
  <div>
    <button
      className="btn btn-success"
      onClick={[Function]}
    >
      Submit
    </button>
    <button
      className="btn btn-secondary"
      onClick={[Function]}
    >
      Clean
    </button>
  </div>
</div>
`;
